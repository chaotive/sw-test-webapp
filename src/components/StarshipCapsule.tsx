import React from 'react'
import {Starship} from '../../typings/swapi/starships'
import Card from 'react-bootstrap/Card'
import {firstCapitalLetter, scaleToPercentaje, zeroOrNone} from '../helpers/formatters'

export interface IStarshipCapsule extends Partial<Starship> {
    name: string
    crew: string
    passengers: string
    hyperdrive_rating: string
}

const hrFormat = (value: string) => value === 'unknown' ? 0 : scaleToPercentaje(value, 0, 5)

const StarshipCapsule: React.FC<IStarshipCapsule> = props => {
  const {name, crew, passengers, hyperdrive_rating} = props
  return <Card className="m-4">
    <Card.Body><strong>Name: {firstCapitalLetter(name)}</strong></Card.Body>
    <Card.Body>Crew: {zeroOrNone(crew)}</Card.Body>
    <Card.Body>Passengers: {zeroOrNone(passengers)}</Card.Body>
    <Card.Body>Hyperdrive Class: {hrFormat(hyperdrive_rating)}</Card.Body>
  </Card>
}

export default StarshipCapsule
