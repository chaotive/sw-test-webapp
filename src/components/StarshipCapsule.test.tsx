import React from 'react'
import ReactDOM from 'react-dom'
import StarshipCapsule, {IStarshipCapsule} from './StarshipCapsule'
import renderer from 'react-test-renderer'

const testData: IStarshipCapsule = {
    name: 'testShip',
    crew: '5',
    passengers: 'some',
    hyperdrive_rating: '0.1'
}

it('renders without crashing', () => {

    const div = document.createElement('div')
    ReactDOM.render(<StarshipCapsule {...testData} />, div)
    ReactDOM.unmountComponentAtNode(div)
})

it('renders correctly', () => {
    const tree = renderer
      .create(<StarshipCapsule {...testData} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
})
