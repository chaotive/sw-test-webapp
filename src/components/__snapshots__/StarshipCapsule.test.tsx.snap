// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="m-4 card"
>
  <div
    className="card-body"
  >
    <strong>
      Name: 
      TestShip
    </strong>
  </div>
  <div
    className="card-body"
  >
    Crew: 
    5
  </div>
  <div
    className="card-body"
  >
    Passengers: 
    some
  </div>
  <div
    className="card-body"
  >
    Hyperdrive Class: 
    0.02
  </div>
</div>
`;
